server/src/main/java/io/crate/planner/optimizer/iterative/IterativeOptimizer.java

```diff
@@ -135,7 +135,10 @@ private boolean exploreNode(int group, Context context) {
                     node = transformed;
                     done = false;
                     progress = true;
-                    context.tracer.ruleApplied(rule, context.memo.extract(), context.planStats);
```

Comment:
`context.memo.extract()` traverses the whole logical plan, converting from group references to the actual plans. This should be only done when we capturing the progress for `explain verbose` or logging.